src/dex-ui/pages/dao/CreateADAO/forms/NFTDAOGovernanceForm.tsx

```diff
@@ -0,0 +1,253 @@
+            id: "governance.nft.maxSupply",
+            label: "Max supply",
+            type: "number",
+            placeholder: "Enter Max Supply",
+            register: {
+              ...register("governance.nft.maxSupply", {
+                required: { value: true, message: "Max supply is required." },
+              }),
+            },
+          }}
+          isInvalid={Boolean(errors.governance?.nft?.maxSupply)}
+          errorMessage={errors.governance?.nft?.maxSupply && errors.governance?.nft?.maxSupply?.message}
+        />
+        <Flex direction="row" gap="4" alignItems="flex-end">
+          <FormInput<"governance.nft.supplyKey">
+            flex="5"
+            inputProps={{
+              flex: 5,
+              id: "governance.nft.supplyKey",
+              label: "Supply Key",
+              type: "text",
+              isReadOnly: true,
+              value: supplyKey,
+              placeholder: "Enter Supply Key",
+              register: {
+                ...register("governance.nft.supplyKey", {
+                  required: { value: true, message: "Supply key is required." },
+                }),
+              },
+            }}
+            isInvalid={Boolean(errors.governance?.nft?.supplyKey)}
+            errorMessage={errors.governance?.nft?.supplyKey && errors.governance?.nft?.supplyKey?.message}
+          />
+          <Button key="get-nft-supply-key" onClick={handleGetAccountInfoClick} flex="1" marginBottom="8px">
+            Get Supply Key
+          </Button>
+        </Flex>
+      </DAOFormContainer>
+      <Text textStyle="p large regular">Initial token distribution</Text>
+      <DAOFormContainer>
+        <FormInput<"governance.nft.treasuryWalletAccountId">
+          inputProps={{
+            id: "governance.nft.treasuryWalletAccountId",
+            label: "Treasury wallet account id",
+            type: "text",
+            placeholder: "Enter wallet account id",
+            register: {
+              ...register("governance.nft.treasuryWalletAccountId", {
+                required: { value: true, message: "A treasury account id is required." },
+              }),
+            },
+          }}
+          isInvalid={Boolean(errors.governance?.nft?.treasuryWalletAccountId)}
+          errorMessage={
+            errors.governance?.nft?.treasuryWalletAccountId && errors.governance?.nft?.treasuryWalletAccountId.message
+          }
+        />
+      </DAOFormContainer>
+      <Flex gap="4" alignItems="center">
+        <Notification
+          type={NotficationTypes.WARNING}
+          textStyle="b3"
+          message={`If you create a new NFT you will need to manually enter 
+                    the id of that NFT in this form. NFT details can be found using the 
+                    hashscan link provided after a successful NFT creation.`}
+        />
+        <Button key="create-nft" onClick={createNewToken}>
+          Create New NFT
+        </Button>
+      </Flex>
+      <LoadingDialog
```

Comment:
You can use conditional states in the `LoadingDialog` props to reduce the number of `LoadingDialog` components from 4 to 2.

Reply:
Done